### ISSUE TITLE
`emblaApi.slidesInView()` should use IntersectionObserver

### ISSUE THREAD

**Issue by davidjerleke** [davidjerleke/embla-carousel#558]
### Feature request is related to

- [x] embla-carousel (core package)

### Is your feature request related to an issue?

- When `emblaApi.slidesInView()` is called intensely on scroll, there's a performance hit.

### Describe the solution you'd like

- The current implementation for `slidesInView` was build for the IE11 era. I'd like to get rid of the performance hit and refactor the `slidesInView` code so it's using `IntersectionObserver` instead, now that IE11 is dead. 

⚠️ The `slidesInView` option will be replaced with the IntersectionObserver `threshold` so it will accept both a single number or an array of numbers:

```tsx
const options = {
  slidesInView: IntersectionObserverInit['threshold'] // Which is --> number | number[] 
}
```

**Comment by davidjerleke** [comment]
Released with [v8.0.0-rc12](https://github.com/davidjerleke/embla-carousel/releases/tag/v8.0.0-rc12).

**Comment by keatkeat87** [comment]
I ran into a related issue but wasn’t sure where to ask, and I could only find this similar one.

We often use a slide’s padding-left to create gaps between slides.
However, this directly affects the IntersectionObserver result, since the padding is considered part of the slide’s bounding box.
In reality, the gap shouldn’t count as part of the slide — when only the gap is visible, the slide should not be considered “in view.”
